src/client/components/app/Transactions/RepayPositionTx.tsx

```diff
@@ -695,14 +757,20 @@ export const RepayPositionTx: FC<RepayPositionProps> = (props) => {
 
             buyToken,
             sellToken,
-            sellAmount: targetAmount,
+            sellAmount: sellAmountInWei, // targetAmount
             network: walletNetwork,
           }).then((result) => {
-            console.log('repay modal: trade quote res', result, result?.buyAmount);
             if (result) {
               setIsTrade(true);
               setHaveFetched0x(true);
-              setTokensToBuy(result.buyAmount!);
+              if (
+                repayType.id === 'claim-and-repay' &&
+                BigNumber.from(result.buyAmount).gt(BigNumber.from(selectedPosition.principal))
```

Comment:
it should be totally valid to buy more than the principal. it will just go into lines reserves. Was there an error on the contract when u did this?

Reply:
we need to test. P sure you are right, but lets leave this in until new SC's are deployed